cltk/text_reuse/automata.py

```diff
@@ -0,0 +1,549 @@
+
+        Adding transitions:
+            You can either define the transition table by assigning
+            a nested dic (see further down for example) to the delta
+            parameter at initialization, or manually define each
+            transition
+
+            >>> A.add_transition('q1', '0', 'q2')
+
+            >>> A.add_transition('q2', '1', 'q2')
+
+            >>> A.add_transition('q2', '0', 'q1')
+
+            The equivalent transition trie will be this:
+
+            >>> A.transition
+            {'q1': {'0': 'q2'}, 'q2': {'1': 'q2', '0': 'q1'}}
+
+            You can not define a transition of unrecognized symbols
+
+            >>> A.add_transition('q1', '2', 'q2')
+            Traceback (most recent call last):
+                ...
+            ValueError
+
+        Calling the transition function:
+            To call δ(qi, u), simply call transition_function:
+
+            >>> A.transition_function('q1', '0')
+            'q2'
+
+            The method returns null if the transition is not defined
+
+            >>> A.transition_function('q1', '1')
+
+        Accepted input strings:
+            Determining whether an input belongs to the language
+            recognized by a DFA A only is O(|w|) time, in contrast
+            to the exponential solution of the equivalnet problem
+            using a nondeterministic automaton.
+
+            Since the automaton is simple, we can deduce its language
+            without the need for any calculations: (01*0)*
+
+            Now on to testing
+
+            >>> A.accepted('01000')
+            True
+
+            >>> A.accepted('010011001110')
+            True
+
+            >>> A.accepted('100100')
+            False
+
+        Complete automaton:
+            A complete automaton is well defined for any state and
+            input symbol. By calling complete_automaton you are
+            essentially creating a new terminating state.
+
+            >>> A.complete_automaton()
+
+            Any undefined transition ends up to the newly defined
+            state:
+
+            >>> A.transition_function('q1', '1') == A.term_state
+            True
+
+            Every transition from the terminating state, reffers
+            back to itself such that δ(term, u) = term
+
+            >>> A.transition_function(A.term_state, '0') == A.term_state
+            True
+
+    """
+
+    def __init__(self, Q, S, s, F, delta = False):
+        """
+        :param Q: set: Set of states Q: q0, q1, ... qn
+        :param S: str set: Set of symbols recognized by the automaton
+        :param s: int: The starting state q_s1
+        :param F: int set: Set of accepted states {q'0, ... q'|F|)
+        :param delta: Nq*|S| int transition matrix (defaults to empty transition matrix)
+        """
+
+        self.Q = Q
+        self.S = S
+
+        if s not in self.Q:
+            LOG.error("The specified value is invalid, s must be a member of Q")
+            raise ValueError
```

Comment:
You will want to create an entry in:
https://github.com/cltk/cltk/blob/master/cltk/exceptions.py
perhaps 
class InputError(Exception):
"""Improper input from the user."""
     pass

And write:
raise InputError("The specified value is invalid, s must be a member of Q")

--The reason is thus: ValueError is indistinguishable from a Python language error; since here you are checking input, you wish to signal the users that they are improperly using the API.
Thus, also, when you write tests or load tests, if you see a true ValueError slip through, then you know there's a bug in your code which isn't trapping an API error condition.

Otherwise this is great, it's always good to check values and report discrepancies that would cause erroneous calculations.